viadot/sources/azure_data_lake.py

```diff
@@ -157,19 +157,32 @@ def to_df(
         path = path or self.path
         url = os.path.join(self.base_url, path)
 
-        if url.endswith(".csv"):
-            df = pd.read_csv(
-                url,
-                storage_options=self.storage_options,
-                sep=sep,
-                quoting=quoting,
-                lineterminator=lineterminator,
-                error_bad_lines=error_bad_lines,
-            )
-        elif url.endswith(".parquet"):
-            df = pd.read_parquet(url, storage_options=self.storage_options)
+        if os.path.splitext(os.path.split(path)[1])[1]:
+            if url.endswith(".csv"):
+                df = pd.read_csv(
+                    url,
+                    storage_options=self.storage_options,
+                    sep=sep,
+                    quoting=quoting,
+                    lineterminator=lineterminator,
+                    error_bad_lines=error_bad_lines,
+                )
+            elif url.endswith(".parquet"):
+                df = pd.read_parquet(url, storage_options=self.storage_options)
+            elif url.endswith(".json"):
+                df = pd.read_json(url, storage_options=self.storage_options, lines=True)
+            else:
+                raise ValueError("Only CSV, parquet and JSON formats are supported.")
         else:
-            raise ValueError("Only CSV and parquet formats are supported.")
+            files = self.ls(path)
+            dfs = []
+            for file in files:
+                print(file)
+                df = pd.read_json(
+                    f"az://{file}", storage_options=self.storage_options, lines=True
+                )
```

Comment:
Why do you assume that if it's a directory, then all the files inside it are JSONs and should be read with `pd.read_json()`? This should be checked by us, with the only assumption that all files in the directory should be of the same format (so we can check one file's extension and choose the loading method based on that). Eg.,
```
files = self.ls(path)
first_file_name, first_file_ext = os.path.splitext(files[0])
...
```
(decide based on extension to use `pd.read_csv()`, `pd.read_parquet()`, `dask_dataframe.read_json()`.